meta-balena-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -82,29 +96,38 @@ sync -f "$SYSROOT"
 mv -T "$SYSROOT/hostapps/.new" "$SYSROOT/hostapps/$CONTAINER_ID"
 sync -f "$SYSROOT"
 
+# Mark it as current hostapp
+ln -srf "$SYSROOT/hostapps/$CONTAINER_ID" "$SYSROOT/current.new"
+sync -f "$SYSROOT"
+mv -T "$SYSROOT/current.new" "$SYSROOT/current"
+sync -f "$SYSROOT"
+
 if [ "$hooks" = 1 ]; then
-	# Run the defined hooks in the host OS we update to
+	# Run prehooks from the new OS we are updating to in existing OS environment
+	new_rootfs=$($SYSROOT/current/boot/init -sysroot $SYSROOT)
+	if DURING_UPDATE=1 $new_rootfs/usr/bin/hostapp-update-hooks -b -d "$new_rootfs/etc/hostapp-update-hooks.d/" ; then
+		echo "Pre hooks ran successfully"
```

Comment:
@agherzan 
This prehooks feature breaks backwards HUP. We don't really test/support a HUP to a previous OS. But it is quite useful. Especially in scenarios where a customers application breaks for whatever reason in the new OS.

The reason this breaks is that `$new_rootfs/usr/bin/hostapp-update-hooks` only supports hostapp-update-hooks v1 as we are hupping to a previous OS.

Perhaps the new hooks script should be named `hostapp-update-hooks-v2`. Its presence tested and then run. Otherwise use legacy hup? Another alternative would be to have os-version specific checks..

Reply:
resolved by using `hostapp-update-hooks-v2`